### PR TITLE
Add cron job for auto-updating IERS tables

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

This is an example of how we could use a cron job to auto-update IERS files every e.g month.

Fixes https://github.com/astropy/astropy/issues/12758

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:5]: Is this daily? Does it need to be so frequent?

**Comment by pllim** [comment]
In principle, I agree with this implementation, which would make the update more transparent by putting everything in the same repo. But I remember @bsipocz had some strong objection for bot to create a branch in this main repo, though I don't see anyway to avoid this without having a security risk of storing someone else's token in this repo, which I am against. 🤷

**Comment by Cadair** [comment]
Branches are cheap and trivial to delete, I think it's definitely better to automate this with the main GH token than any other options.

**Comment by astrofrog** [comment]
Well one option would be to make a dedicated astropy bot GitHub account we could fork to if people preferred (similar to astrobot)

**Comment by bsipocz** [comment]
There is already a bot account, what about reusing it?

**Comment by astrofrog** [comment]
I think astropy-bot was never a full account just a GitHub app but I will double check.

**Comment by pllim** [comment]
@Cadair and I are still of the opinion that it is no big deal to have a short-lived branch right here on this repo for a few times the IERS table gets updated. That way, we don't need to worry about third-party bot accounts.

**Comment by Cadair** [comment]
:+1: I think it's no where near worth the complexity of dealing with forks to worry about it.

**Comment by bsipocz** [comment]
> short-lived branch

if this is true, I agree. Otherwise don't, people fork and end up with weird stuff, I had more than my fair share of cleanup of stray branches and tags creeping back to the main repo over and over again thanks to non-clean forks of people...

**Comment by mhvk** [comment]
This looks very nice! 

The main question indeed seems to be whether to use a standard bot-fork or work with the main repo. In principle, in `git` there should be no difference between one repo or a fork, but github often decides differently. So, if there is a separate bot fork already, it would seem to make sense to use it.

p.s. Right now, the main archive has quite a few extraneous ones:
```
upstream/dhomeier-fix-bbox-mpl2 
upstream/LTS 
upstream/main 
upstream/older-docs-archive 
upstream/revert-11508-fix-3972 
upstream/revert-11771-revert-11046-update_spinner_keep_enum 
upstream/revert-12567-refactor-distance 
upstream/stable 
upstream/taldcroft-patch-1 
upstream/v0.1.x 
upstream/v0.2.x 
upstream/v0.3.x 
upstream/v0.4.x 
upstream/v1.0.x 
upstream/v1.1.x 
upstream/v1.2.x 
upstream/v1.3.x 
upstream/v2.0.x 
upstream/v3.0.x 
upstream/v3.1.x 
upstream/v3.2.x 
upstream/v4.0.x 
upstream/v4.1.x 
upstream/v4.2.x 
upstream/v4.3.x 
upstream/v5.0.x 
```

File: .github/workflows/update_iers.yml
Comment (mhvk) [reply .github/workflows/update_iers.yml:5]: Daily does seem overkill since really this is needed only to keep any release roughly up to date. Ideally, it "knows" when a release is imminent and only then triggers...  But otherwise, monthly is probably good enough.

**Comment by Cadair** [comment]
> I had more than my fair share of cleanup of stray branches and tags creeping back to the main repo over and over again thanks to non-clean forks of people

Tags are a bit different, but people pushing tags or branches direct to upstream is a whole other level, people with push rights should be able to use git better than that!

>  So, if there is a separate bot fork already, it would seem to make sense to use it.

It's not this straight forward, if we use a different account and fork based workflow the maintainability of this action drops dramatically, as you have to deal with who has access to the bot account, where does the workflow live, and depending on where it lives who has the access tokens etc.

It just seems like we are shooting ourselves in the foot for absolutely no good reason. Having stray branches lying around hurts nobody unless they have too much time on their hands to be worrying about how long a list of branches is somewhere. Not to mention this wont even make stray branches for longer than the review time on an automatically updated data file (which should be approximately one run of the CI).

**Comment by mhvk** [comment]
@Cadair - thanks, that sounds encouraging. And good point about maintenance of the bot being much easier in the main account.

**Comment by eerovaher** [comment]
> p.s. Right now, the main archive has quite a few extraneous ones:

@mhvk, there aren't any unnecessary branches at the moment in the GitHub repository, see Git documentation for why they are still present in your local repository: https://git-scm.com/docs/git-fetch#_pruning

GitHub provides a button in its user interface to revert a merged pull request. The branches with names `revert-XXXXX-...` were created by GitHub when that button was pressed, they were not pushed from a personal repository.

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy/pull/12882#issuecomment-1050772689 -- @mhvk , I suspect the `revert` ones will go away when you go `git remote prune upstream` locally.

**Comment by mhvk** [comment]
Thanks, @pllim, they indeed disappeared.  Learn something new every day!

**Comment by bsipocz** [comment]
>>    I had more than my fair share of cleanup of stray branches and tags creeping back to the main repo over and over again thanks to non-clean forks of people

> Tags are a bit different, but people pushing tags or branches direct to upstream is a whole other level, people with push rights should be able to use git better than that!

🤣

**Comment by Cadair** [comment]
**should** :wink:

**Comment by pllim** [comment]
Let's focus back on this particular topic. This is going to only affect `astropy` core. I still think having a short-lived branch here is not a problem.

**Comment by astrofrog** [comment]
Ok so I've updated this, changes are now:

* Create branches in the core repo, these can be deleted when PRs are merged
* Remove use of batchpr, this can just be done with a few steps in an action workflow and using the widely used https://github.com/peter-evans/create-pull-request action
* Run weekly for now, we can update to monthly once it works well? (otherwise debugging will be veeeery slow)
* Use astrobot account to commit

I think this is ready to go and I tested that it works on my fork, e.g. https://github.com/astrofrog/astropy/pull/106

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:29]: So in the release procedure, there needs to be reminder to update the label here every LTS release?

And can we set the milestone automatically too?

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:27]: Is the trailing `-` necessary?

```suggestion
        branch: update-iers
```

**Review by pllim — COMMENTED** [review]
Looking good! Some minor comments.

File: .github/workflows/update_iers.yml
Comment (astrofrog) [reply .github/workflows/update_iers.yml:29]: Well for milestone the issue is that would need to be updated quite often as it would be different for each minor release. I'm tempted to also remove the backport labelling because otherwise it's yet another manual step during releases. Maybe it's enough to rely on maintainers labelling/milestoning the PR? (aside from the other non-backport labels above).

File: .github/workflows/update_iers.yml
Comment (astrofrog) [reply .github/workflows/update_iers.yml:20]: Good idea

Comment (astrofrog) [reply .github/workflows/update_iers.yml]: Done

File: docs/development/releasing.rst
Comment (astrofrog) [reply docs/development/releasing.rst:130]: I'll have a think about how to mention this here while still decluttering the main release notes

Comment (pllim) [reply docs/development/releasing.rst:130]: Please ping me again when this is ready for another review. Thanks!

**Comment by astrofrog** [comment]
@pllim - I've now added back a mention to the docs under the CI checks (since it is basically just to check the CI has run ok for the auto-update). I've also now not labelled for backport automatically but asked maintainers to do this since otherwise the CI config will need to be updated at each major release manually (not just LTS).

**Comment by astrofrog** [comment]
I've also now added an option that should I think auto-delete the branch on merging

**Comment by astrofrog** [comment]
@pllim - I think this is now ready for review!

**Comment by pllim** [comment]
Maybe squash? 😬

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:8]: Accidental indentation?
```suggestion
jobs:
```

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:18]: So, is this smart enough to not create a PR if there is no change?

File: docs/development/releasing.rst
Comment (pllim) [inline docs/development/releasing.rst:228]:
```suggestion
successfully run and that related pull requests have been merged (and backported if needed).
```

**Review by pllim — COMMENTED** [review]
I think we're very close. Did you test this again in your own fork?

File: docs/development/releasing.rst
Comment (saimn) [inline docs/development/releasing.rst:257]: Why is this block deleted ? (It doesn't seem related to this PR)

Comment (astrofrog) [reply docs/development/releasing.rst:257]: It is no longer needed but I forgot to remove it in the PR that updated the CI config (https://github.com/astropy/astropy/pull/12832) - just thought I'd remove it here since I was editing the release docs anyway.

File: .github/workflows/update_iers.yml
Comment (astrofrog) [reply .github/workflows/update_iers.yml:17]: It is now, I made the ``git commit`` conditional, and the PR action is smart enough to see if there are no new commits.

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:5]: Did you change this from weekly to monthly?

**Comment by astrofrog** [comment]
I've tested this on my fork and it works well. I've also squashed down to 5 commits and set it to run monthly. I think this is good to go!

File: .github/workflows/update_iers.yml
Comment (astrofrog) [reply .github/workflows/update_iers.yml:5]: Yes, because I can now use the worklow dispatch for debugging if needed, so I can just set the cron to what we really want.

File: .github/workflows/update_iers.yml
Comment (pllim) [inline .github/workflows/update_iers.yml:32]: Nice. Best of both worlds!

**Review by pllim — APPROVED** [review]
Looks pretty clean now. Let's deploy and see what happens... 🙈 

Thanks!

**Comment by pllim** [comment]
I'll let @saimn have one more look.

**Review by saimn — APPROVED** [review]
LGTM, seems a nice and simple way to solve the issue.  Thanks @astrofrog !

**Pull request merged**
